src/NITTA/Intermediate/DataFlow.hs

```diff
@@ -88,3 +95,71 @@ fsToDataFlowGraph fs = DFCluster $ map DFLeaf fs
 
 addFuncToDataFlowGraph f (DFCluster items) = DFCluster (DFLeaf f : items)
 addFuncToDataFlowGraph f leaf = DFCluster [DFLeaf f, leaf]
+
+dataFlowGraphToFs (DFLeaf f) = [f]
+dataFlowGraphToFs (DFCluster dfs) = foldl (\fs df -> dataFlowGraphToFs df ++ fs) [] dfs
+
+data ProcessFlowStep v x = ProcessFlowStep
```

Comment:
Naming and description. Also, see on `bindWaves` metric, which should be similar. We should avoid code duplication.

Also see `reorderAlgorithm`